app/[locale]/(interactive)/4-6/page.tsx

```diff
@@ -0,0 +1,145 @@
+    if (!stroke.length) return "";
+    const d = stroke.reduce(
+      (acc: any[], [x0, y0]: any, i: number, arr: string | any[]) => {
+        const [x1, y1] = arr[(i + 1) % arr.length];
+        acc.push(x0, y0, (x0 + x1) / 2, (y0 + y1) / 2);
+        return acc;
+      },
+      ["M", ...stroke[0], "Q"],
+    );
+    d.push("Z");
+    return d.join(" ");
+  }
+
+  const [hasWrite, setWrite] = React.useState<boolean>(false);
+  const currentStroke = useRef<number[][]>([]);
+  const svgRef = useRef();
+
+  function handlePointerDown(e: React.PointerEvent<SVGSVGElement>) {
+    setWrite(true);
+    e.currentTarget.setPointerCapture(e.pointerId);
+    const pos = svgRef.current.getBoundingClientRect();
+    currentStroke.current = [
+      [e.pageX - pos.left, e.pageY - pos.top, e.pressure],
+    ];
+  }
+
+  function handlePointerMove(e: React.PointerEvent<SVGSVGElement>) {
+    if (e.buttons !== 1) return;
+    const pos = svgRef.current.getBoundingClientRect();
+    currentStroke.current.push([
+      e.pageX - pos.left,
+      e.pageY - pos.top,
+      e.pressure,
+    ]);
+    updateNameStorke([...nameStorke]);
+  }
+
+  function handlePointerUp() {
+    // new function to handle new independent stroke
+    updateNameStorke([...nameStorke, currentStroke.current]);
+    currentStroke.current = [];
+  }
+
+  function clearStrokes() {
+    setWrite(false);
+    updateNameStorke([]);
+  }
+
+  const renderedStrokes = nameStorke.map((stroke, index) => {
+    const pathData = getSvgPathFromStroke(
+      getStroke(stroke, {
+        size: 8,
+        thinning: 0.5,
+        smoothing: 0.5,
+        streamline: 0.5,
+      }),
+    );
+    return <path key={index} d={pathData} />;
+  });
+
+  return (
+    <>
+      <div className="flex h-full w-full flex-col items-center justify-center">
+        <div className="flex flex-col items-center space-y-5 text-center text-[26px]">
+          <p className="text-[#F8F8F7]">เจ้าของสมุดเล่มนี้คือ...</p>
+          <div className="relative">
+            <svg
+              id="svg"
+              ref={svgRef}
+              onPointerDown={handlePointerDown}
+              onPointerMove={handlePointerMove}
+              onPointerUp={handlePointerUp}
+              className="relative h-[158px] w-[331px] touch-none rounded-[10px] bg-[#F8F8F7] opacity-80"
+            >
+              {renderedStrokes}
+              {currentStroke.current.length > 0 && (
+                <path
+                  d={getSvgPathFromStroke(
+                    getStroke(currentStroke.current, {
+                      size: 8,
+                      thinning: 0.5,
+                      smoothing: 0.5,
+                      streamline: 0.5,
+                    }),
+                  )}
+                />
+              )}
+            </svg>
+
+            {!hasWrite && (
+              <p className="absolute left-[59px] top-[59px] text-[#A98F7E]">
+                ลากนิ้วเพื่อเขียนชื่อ
```

Comment:
this is not i18n-ized too

Reply:
Page 4-6, 4-7 should be in [this branch](https://github.com/CBC-Warehouse-Of-Heart/Warehouse-of-Heart/tree/feat/4-1_4-2). I recently pushed my codes for managing perfect-freehand strokes in that branch.